In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [5]:
players <- read_csv("players.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [6]:
sessions <- read_csv("sessions.csv")

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Question 1: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

(1) Data Description: The players contains 196 observations and 7 variables, including exprience,hashedEmail,name,gender,played_hours,age and subscribe. The key reposonse variable is subscribe, it is the logical variable indicating whether the player subscribed to a game-related newsletter. Predictor variables include numerical features like age and played_hours. The dataset may contain missing or inconsistent entries, and the subscription rate may be unbalanced. The sessions contains 1535 observations and 5 variables: hashedEmail,start_time,end_time,original_start_time,original_end_time. It records session level activity, which can be used to calculate total playtime, session frequency and active hours. The two dataset both have hashedEmail, so they can be linked by using this variable. 

(2)Questions: Broad Question: what player characteristics and behaviors are most predictable of subscribing to a game-related newsletter? 

Specific question: Can player characteristics like age, experience and total played hours predict whether a player subscribes to the newsletter? 

The analysis will use the players.csv dataset and may be combined with the aggregated session information from the sessions.csv file. The goal is to determine which features are closely related to the likelihood of subscription.